sscanss/app/dialogs/misc.py

```diff
@@ -1369,3 +1371,325 @@ def plot(self):
+        new_index = inv_index[selected_index]
+
+        if new_index > selected_index:
+            start = selected_index if len(self.inputs) > len(intensity) else selected_index + 1
+            intensity = np.delete(self.inputs, slice(start, new_index + 1))
+            alpha = np.delete(self.outputs, slice(start, new_index + 1))
+        elif new_index < selected_index:
+            intensity = np.delete(self.inputs, slice(new_index, selected_index))
+            alpha = np.delete(self.outputs, slice(new_index, selected_index))
+            selected_index = new_index
+
+        self.inputs = intensity
+        self.outputs = alpha
+        self.selected_index = selected_index
+
+        return selected_index
+
+    def canvasMousePressEvent(self, event):
+        """Matplotlib canvas mouse press event handler
+
+        :param event: mouse event
+        :type event: matplotlib.backend_bases.MouseEvent
+        """
+        if event.button != 1 or event.inaxes is None:
+            return
+
+        selected_index = self.getSelectedPoint(event)
+        if selected_index is None:
+            x = max(min(trunc(event.xdata, 2), self.edge[-1]), self.edge[0])
+            y = max(min(trunc(event.ydata, 2), 1.0), 0.0)
+            self.inputs = np.append(self.inputs, x)
+            self.outputs = np.append(self.outputs, y)
+            self.inputs, index, inv_index = np.unique(self.inputs, return_index=True, return_inverse=True)
+            self.outputs = self.outputs[index]
+            self.selected_index = inv_index[-1]
+        else:
+            self.selected_index = selected_index
+        self.last_pos = self.selected_index
+        self.createCurve()
+
+    def canvasMouseReleaseEvent(self, event):
+        """Matplotlib canvas mouse release event handler
+
+        :param event: mouse event
+        :type event: matplotlib.backend_bases.MouseEvent
+        """
+        if event.button != 1:
+            return
+
+        self.last_pos = None
+
+    def canvasMouseMoveEvent(self, event):
+        """Matplotlib canvas mouse move event handler
+
+        :param event: mouse event
+        :type event: matplotlib.backend_bases.MouseEvent
+        """
+
+        if event.inaxes is None:
+            return
+
+        x = max(min(trunc(event.xdata, 2), self.edge[-1]), self.edge[0])
+        y = max(min(trunc(event.ydata, 2), 1.0), 0.0)
+        self.cursor_pos.set_text(f'({x:.2f}, {y:.2f})')
+
+        if self.last_pos is None or event.button != 1:
+            self.canvas.draw()
+            return
+
+        self.last_pos = self.__updateAndMerge(x, y, self.last_pos)
+
+        self.createCurve()
+
+    def createCurve(self):
+        """Creates a new Curve object with selected inputs"""
+        self.curve = Curve(self.inputs, self.outputs, (self.edge[0], self.edge[-1]),
+                           Curve.Type(self.type_combobox.currentText()))
+        self.plot()
+        self.previewCurve(self.curve)
```

Comment:
I find the UI feels more responsive if I move previewCurve to only be called in canvasMouseReleaseEvent.